.changes/unreleased/BUG FIXES-20240719-143606.yaml

```
@@ -0,0 +1,6 @@
+kind: BUG FIXES
+body: 'data-source/http: allow the `Host` request header to be sent as the http request
+  host.'
```

```suggestion
body: 'data-source/http: Allow `Host` header in `request_headers` to be set on HTTP request'
```

nit: slight rewording